### ISSUE TITLE
Change input field selection to be role-based

### ISSUE THREAD

**Issue by joedolson** [joedolson/my-calendar#248]
I'm wondering if I should completely re-map this feature [event input field selection] so that it's more related to permissions. E.g., set up these input fields for admins, these for editors, etc. That would allow much more control over everything. It would be a big change, though. Break some backwards compatibility, for sure.

**Comment by paaljoachim** [comment]
I am not sure I see the reason why it is needed. What is an example use case?

**Comment by joedolson** [comment]
For example, would be "Editors can add new locations, but not Contributors" "Only Admins can set up recurring events"

It might be something that's only suitable as a filter, not as UI.

**Comment by paaljoachim** [comment]
I thought we had something similar in the Settings -> Permissions screen.....

**Comment by joedolson** [comment]
Yes, but it doesn't control specific input fields. (Actually, that does provide controls for adding categories or locations, but not for all the other fields.)
